### PR TITLE
chore: Add travis-ci config and badge

### PR DESCRIPTION
**Pull request by pgoldrbx**
## Description
Add travis-ci configuration and status badge
https://travis-ci.org/CondeNast/xml-to-react

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore, documentation, cleanup

### CODE REVIEW

**Comment by pgoldrbx** [comment]
After merging we can make this a required check for PRs

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+language: node_js
+node_js:
+  - "6"
```
Comment (gautamarora) [inline .travis.yml]: why run travis on node v6? could we skip this and let it fallback to using node version from `.nvmrc`?
https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#Specifying-Node.js-versions-using-.nvmrc

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  - "8"
+
+notifications:
+  email: false
+
+install:
+  - npm install
+
+script:
+  - npm test
```
Comment (gautamarora) [inline .travis.yml]: should we mention this, or let it fallback to default? https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#Default-Build-Script

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+language: node_js
+node_js:
+  - "6"
+  - "8"
+
+notifications:
+  email: false
+
+install:
+  - npm install
```
Comment (gautamarora) [inline .travis.yml]: is this needed or will fallback to `npm install` as a default for node project?

File: .travis.yml (hunk 1 above)
Comment (pgoldrbx) [reply .travis.yml]: sure, we can, but we've just set our nvmrc for local development. we can also take advantage of supporting node LTS for the module itself, regardless of what version is used for development.

File: .travis.yml (hunk 2 above)
Comment (pgoldrbx) [reply .travis.yml]: i tend to be deliberate but we could go w/ implicit

Comment (gautamarora) [reply .travis.yml]: sure, being explicit works. i wanted to present the option :)

File: .travis.yml (hunk 1 above)
Comment (gautamarora) [reply .travis.yml]: yes, sounds good on having this here. thanks for clarifying :)

**Pull request merged**